core/src/main/java/org/elasticsearch/ElasticsearchException.java

```diff
@@ -702,4 +689,25 @@ public static void renderThrowable(XContentBuilder builder, Params params, Throw
         ElasticsearchException.toXContent(builder, params, t);
         builder.endObject();
     }
+
+    interface IOFunction<T, R> {
+
+        /**
+         * Applies this function to the given argument.
```

Comment:
This seems a bit meaningless. Maybe a javadoc on the interface about how its built rather than on apply?